x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Alias.java

```diff
@@ -147,4 +152,20 @@ public String nodeString() {
     public static Expression unwrap(Expression e) {
         return e instanceof Alias as ? as.child() : e;
     }
+
+    @Override
+    @SuppressWarnings("checkstyle:EqualsHashCode")// equals is implemented in parent. See innerEquals instead
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), id());
+    }
+
+    /**
+     * Compared to e.g. {@link UnresolvedAttribute}, the id is part of equality for Alias because the {@link ReferenceAttribute} created
+     * by {@link #toAttribute()} uses it.
```

Comment:
```suggestion
     * The id is part of equality for Alias because the {@link ReferenceAttribute} created
     * by {@link #toAttribute()} uses it.
```